tpl/reflect/reflect_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reflect
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var ns = New()
+
+type tstNoStringer struct{}
+
+func TestIsMap(t *testing.T) {
+	for i, test := range []struct {
+		v      interface{}
+		expect interface{}
+	}{
+		{map[int]int{1: 1}, true},
+		// failures
+		{nil, false},
+		{"foo", false},
+		{1.0, false},
+		{true, false},
+		{[]int{1, 2}, false},
+		{tstNoStringer{}, false},
+		{&tstNoStringer{}, false},
+	} {
+		errMsg := fmt.Sprintf("[%d] %v", i, test)
+		result := ns.IsMap(test.v)
+		assert.Equal(t, test.expect, result, errMsg)
+	}
+}
+
+func TestIsSlice(t *testing.T) {
+	for i, test := range []struct {
+		v      interface{}
+		expect interface{}
+	}{
+		{[]int{1, 2}, true},
+		{[]tstNoStringer{{}, {}}, true},
+		// failures
+		{nil, false},
```

Comment:
I have only one comment to this PR. The failures section:

1. It's not failures. There is no error involved.
2. It looks copy-pasted from somewhere else. I think it's fair to assume that Go's reflect package is well tested, so any tests in here should either test the integration and possibly also document the usage. So I would say that 1 slice (true) and 2 falses (nil + some other) would be enough and easier to understand. As it is now, it looks way more complex than it is. This comment also relates to the map variant.

Reply:
PTAL